src/HotChocolate/Data/src/EntityFramework/Pagination/EfQueryableCursorPagingHandler.cs

```diff
@@ -103,14 +103,15 @@ private async ValueTask<Connection> SliceAsync(
                 .ToAsyncEnumerable(context.RequestAborted)
                 .ConfigureAwait(false))
             {
-                builder.Add(new Edge<TEntity>(item.Item, CursorFormatter.Format(item.Item, keys)));
-                totalCount ??= item.TotalCount;
                 fetchCount++;
 
-                if (fetchCount >= requestedCount)
```

Comment:
the actuall issue was this check

Reply:
Yes, indeed. If the only change would've been `if (fetchCount > requestedCount)` without altering the order (incrementing the count first and then adding the entities), it might result in reading one additional entry unnecessarily (besides my obvoius mistake with continue previously which made it worse).